x-pack/plugins/apm/public/components/app/Main/useUpdateBadgeEffect.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useEffect } from 'react';
+import { capabilities } from 'ui/capabilities';
+import chrome from 'ui/chrome';
+
+export const useUpdateBadgeEffect = () => {
+  useEffect(() => {
+    const uiCapabilities = capabilities.get();
+    chrome.badge.set(
+      !uiCapabilities.apm.save
+        ? {
+            text: i18n.translate('xpack.apm.header.badge.readOnly.text', {
+              defaultMessage: 'Read only'
+            }),
+            tooltip: i18n.translate('xpack.apm.header.badge.readOnly.tooltip', {
+              defaultMessage: 'Unable to save'
+            }),
+            iconType: 'glasses'
+          }
+        : undefined
+    );
+  }, []);
+};
```

Comment:
Is this component specific to APM? If not, perhaps it should be in `ui/capabilities`.

Reply:
The read-only badges are currently application specific